### PR TITLE
Presentation: Final docs review

### PR DESCRIPTION
**Pull request by grigasp**
TODO:
- [x] Cross-link rule and specification markdowns
- [x] Add attribute information tables
- [x] Unify & cross-link markdowns and TS docs
- [x] Remove `.snag` files
- [x] Make all presentation docs sub-folders lowercase

### CODE REVIEW

**Comment by grigasp** [comment]
@iTwin/itwinjs-core-presentation, this is ready for review.

One thing I I'm not sure about is whether we need to link each attribute to its markdown docs. At the moment I only link interfaces. Would be happy to hear your thoughts on this.

File: presentation/common/src/presentation-common/rules/customization/SortingRule.ts [hunk 1]
```diff
@@ -7,75 +7,62 @@
...
 }
 
 /**
- * Rule to disable sorting for certain ECInstances in the hierarchy and/or content.
- *
- * **Note:** Disabling sorting increases performance
- *
- * **Note:** This rule is not meant to be used to sort grouping nodes, custom nodes or
- * other non ECInstance type of nodes.
+ *R ule to disable sorting for certain ECInstances in the hierarchy and/or content.
```
Comment (roluk) [inline presentation/common/src/presentation-common/rules/customization/SortingRule.ts:63]:
```suggestion
 * Rule to disable sorting for certain ECInstances in the hierarchy and/or content.
```

**Review by roluk — CHANGES_REQUESTED** [review]

File: docs/presentation/hierarchies/CustomNode.md [hunk 2]
```diff
@@ -148,6 +177,13 @@ empty list. It's up to the UI component to handle the case of parent node saying
...
 list even if there are child node rules that define children for it.
 
+|                      |                                                                                                                                                      |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Type**             | `"Always" \| "Never" \| "Unknown"`                                                                                                                   |
+| **Is Required**      | No                                                                                                                                                   |
+| **Default Value**    | `"Unknown"`                                                                                                                                          |
+| **Performance note** | Setting the attribute to `Always` or `Never` may substantially improve performance of creating nodes in cases when  getting child nodes is expensive |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/presentation/hierarchies/CustomNode.md]: Made some changes, please check https://github.com/iTwin/itwinjs-core/pull/3388/commits/4f258d42bc5560b047bdad458ac08c4f104f6d8a

File: docs/presentation/hierarchies/InstanceNodesOfSpecificClasses.md [hunk 3]
```diff
@@ -129,12 +162,15 @@ Controls the order in which specifications are handled — specification with hi
...
-> **Performance note:** Setting the attribute may substantially improve performance when creating large numbers of nodes.
-
 Suppress sorting of nodes returned by this specification. With this attribute set to `true`, the order of returned nodes is undefined.
 
+|                      |                                                                                                            |
+| -------------------- | ---------------------------------------------------------------------------------------------------------- |
+| **Type**             | `boolean`                                                                                                  |
+| **Is Required**      | No                                                                                                         |
+| **Default Value**    | `false`                                                                                                    |
+| **Performance note** | Setting the attribute to `true` may substantially improve performance when creating large numbers of nodes |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/presentation/hierarchies/InstanceNodesOfSpecificClasses.md]: Made some changes, please check https://github.com/iTwin/itwinjs-core/pull/3388/commits/4f258d42bc5560b047bdad458ac08c4f104f6d8a

File: docs/presentation/hierarchies/InstanceNodesOfSpecificClasses.md [hunk 4]
```diff
@@ -193,6 +232,13 @@ empty list. It's up to the UI component to handle the case of parent node saying
...
 list even if there are child node rules that define children for it.
 
+|                      |                                                                                                                                                      |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Type**             | `"Always" \| "Never" \| "Unknown"`                                                                                                                   |
+| **Is Required**      | No                                                                                                                                                   |
+| **Default Value**    | `"Unknown"`                                                                                                                                          |
+| **Performance note** | Setting the attribute to `Always` or `Never` may substantially improve performance of creating nodes in cases when  getting child nodes is expensive |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/presentation/hierarchies/InstanceNodesOfSpecificClasses.md]: Made some changes, please check https://github.com/iTwin/itwinjs-core/pull/3388/commits/4f258d42bc5560b047bdad458ac08c4f104f6d8a

File: docs/presentation/hierarchies/RelatedInstanceNodes.md [hunk 5]
```diff
@@ -116,12 +145,15 @@ Controls the order in which specifications are handled — specification with hi
...
-> **Performance note:** Setting the attribute may substantially improve performance when creating large numbers of nodes.
-
 Suppress sorting of nodes returned by this specification. With this attribute set to `true`, the order of returned nodes is undefined.
 
+|                      |                                                                                                            |
+| -------------------- | ---------------------------------------------------------------------------------------------------------- |
+| **Type**             | `boolean`                                                                                                  |
+| **Is Required**      | No                                                                                                         |
+| **Default Value**    | `false`                                                                                                    |
+| **Performance note** | Setting the attribute to `true` may substantially improve performance when creating large numbers of nodes |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/presentation/hierarchies/RelatedInstanceNodes.md]: Made some changes, please check https://github.com/iTwin/itwinjs-core/pull/3388/commits/4f258d42bc5560b047bdad458ac08c4f104f6d8a

File: docs/presentation/hierarchies/RelatedInstanceNodes.md [hunk 6]
```diff
@@ -180,6 +215,13 @@ empty list. It's up to the UI component to handle the case of parent node saying
...
 list even if there are child node rules that define children for it.
 
+|                      |                                                                                                                                                      |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Type**             | `"Always" \| "Never" \| "Unknown"`                                                                                                                   |
+| **Is Required**      | No                                                                                                                                                   |
+| **Default Value**    | `"Unknown"`                                                                                                                                          |
+| **Performance note** | Setting the attribute to `Always` or `Never` may substantially improve performance of creating nodes in cases when  getting child nodes is expensive |
```
[the remark this answers is not in the document]
Comment (grigasp) [reply docs/presentation/hierarchies/RelatedInstanceNodes.md]: Made some changes, please check https://github.com/iTwin/itwinjs-core/pull/3388/commits/4f258d42bc5560b047bdad458ac08c4f104f6d8a

File: docs/presentation/hierarchies/CustomNode.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (roluk) [reply docs/presentation/hierarchies/CustomNode.md]: Looks much better now, thanks.

**Comment by roluk** [comment]
You can mark this PR ready now.

> One thing I'm not sure about is whether we need to link each attribute to its markdown docs. At the moment I only link interfaces. Would be happy to hear your thoughts on this.

By now, documentation of specification attributes have so much information, that they each deserve separate documentation pages, just like TypeScript class members. Until this split is made, I would not bother linking each member to their respective documentation page.

**Review by roluk — APPROVED** [review]

**Pull request merged**